x-pack/filebeat/input/s3/input.go

```diff
@@ -340,7 +340,7 @@ func handleSQSMessage(m sqs.Message) ([]s3Info, error) {
 
 	var s3Infos []s3Info
 	for _, record := range msg.Records {
-		if record.EventSource == "aws:s3" && record.EventName == "ObjectCreated:Put" {
+		if record.EventSource == "aws:s3" && strings.HasPrefix(record.EventName, "ObjectCreated:") {
```

Comment:
How about we use the constant `EventS3ObjectCreated` here from aws-sdk-go-v2 https://github.com/aws/aws-sdk-go-v2/blob/master/service/s3/api_enums.go#L2?

Reply:
> How about we use the constant `EventS3ObjectCreated` here from aws-sdk-go-v2 https://github.com/aws/aws-sdk-go-v2/blob/master/service/s3/api_enums.go#L2?

The enums there is the full event name including the `s3:` prefix. However, the raw event data looks like this:

```
{
   "Records":[ 
      { 
         "eventVersion":"2.1",
         "eventSource":"aws:s3",
         "awsRegion":"eu-central-1",
         "eventTime":"2019-10-16T18:20:33.017Z",
         "eventName":"ObjectCreated:CompleteMultipartUpload",
         "userIdentity":{ 
            "principalId":"AWS:123"
         },
         "requestParameters":{ 
            "sourceIPAddress":"0.0.0.0"
         },
         "responseElements":{ 
            "x-amz-request-id":"ABCD",
            "x-amz-id-2":"abc="
         },
         "s3":{ 
            "s3SchemaVersion":"1.0",
            "configurationId":"xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
            "bucket":{ 
               "name":"testbucket",
               "ownerIdentity":{ 
                  "principalId":"ABCD"
               },
               "arn":"arn:aws:s3:::testbucket"
            },
            "object":{ 
               "key":"accesslog_20191009-132000-00.gz",
               "size":8192,
               "eTag":"abcd",
               "versionId":"abcd",
               "sequencer":"AADA50XXXX"
            }
         }
      }
   ]
}

Which means that we cannot use the enums directly unless we use a substring match or string manipulation. That's why I recommend using HasPrefix.
```

Reply:
Ahh good point, didn't see the `s3:` prefix. Thanks!

Reply:
@jsoriano Since we can't use the enum directly, checking the prefix sounds good to me. WDYT?

Reply:
Sounds great, thanks!